### PR TITLE
Trigger a hot reload / full restart based on SIGUSR1 and SIGUSR2

### PR DESCRIPTION
**Pull request by johnmccutchan**
- [x] Remove --control-pipe command option and related code. Reading from a named pipe via dart:io is unreliable today.
- [x] Add --pid-file option which writes the pid of flutter run into a file.
- [x] Wired up SIGUSR1 to hot reload
- [x] Wired up SIGUSR2 to full restart

### CODE REVIEW

**Comment by johnmccutchan** [comment]
@Hixie @HansMuller

**Comment by johnmccutchan** [comment]
Expected usage:

```
$ flutter run --hot --pid-file=/tmp/flutter.run.pid
```

To trigger a hot reload from another process:

```
$ kill -USR1 `cat /tmp/flutter.run.pid`
```

To trigger a restart from another process:

```
$ kill -USR2 `cat /tmp/flutter.run.pid`
```

**Comment by HansMuller** [comment]
LGTM

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 1]
```diff
@@ -65,6 +65,14 @@ abstract class ResidentRunner {
       await cleanupAfterSignal();
       exit(0);
     });
+    ProcessSignal.SIGUSR1.watch().listen((ProcessSignal signal) async {
+      printStatus('Caught SIGUSR1');
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/resident_runner.dart]: This should say what's going to happen next, e.g. "Caught SIGUSR1 - attempting hot reload".

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 2]
```diff
@@ -65,6 +65,14 @@ abstract class ResidentRunner {
...
+      await restart(fullRestart: false);
+    });
+    ProcessSignal.SIGUSR2.watch().listen((ProcessSignal signal) async {
+      printStatus('Caught SIGUSR2');
+      await restart(fullRestart: true);
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/resident_runner.dart]: ditto

File: packages/flutter_tools/lib/src/commands/run.dart [hunk 3]
```diff
@@ -69,10 +68,9 @@ class RunCommand extends RunCommandBase {
                       defaultsTo: false,
                       help: 'Run with support for hot reloading.');
 
-    // Option to enable control over a named pipe.
-    argParser.addOption('control-pipe',
-                        hide: true,
-                        help: 'Specify a named pipe to receive commands on.');
+    // Option to write the pid to a file.
+    argParser.addOption('pid-file',
+                        help: 'Specify a file to write the process id to.');
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/run.dart]: This should mention SIGUSR1 and SIGUSR2.

**Comment by Hixie** [comment]
LGTM

File: packages/flutter_tools/lib/src/resident_runner.dart (hunk 1 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/resident_runner.dart]: The hot reload code that we call immediately prints that it is doing a hot reload.

File: packages/flutter_tools/lib/src/resident_runner.dart (hunk 2 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/resident_runner.dart]: ditto

File: packages/flutter_tools/lib/src/commands/run.dart (hunk 3 above)
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/commands/run.dart]: Done (in a separate CL).

**Pull request merged**